blueocean-core-js/src/js/User.js

```diff
@@ -21,6 +98,18 @@ export class User {
     static current() {
         return CURRENT; // eslint-disable-line no-use-before-define
     }
+
+    get isAdministrator() {
+        const permission = this._blueUser._permission || {};
+        return !!permission.administrator;
+    }
+
+    get permissions() {
+        return {
+            pipeline: new PipelinePermissions(this._blueUser),
+            credential: new CredentialPermissions(this._blueUser),
+        };
+    }
```

Comment:
Do we want to add a getter or function for every single property in this graph? It seems like a maintenance problem as new permissions are added, we need to keep adding them here, and republishing every time. Should we consider doing a deep clone from the passed in `blueUser` directly onto the `User` instance?